src/requestor/routes/maintain.py

```diff
@@ -198,27 +224,36 @@ async def update_request(
 
         # the access request is approved: grant access
         if status in config["UPDATE_ACCESS_STATUSES"]:
+            action = "revoke" if request.revoke else "grant"
             logger.debug(
-                f"Status is one of '{config['UPDATE_ACCESS_STATUSES']}', attempting to grant access in Arborist"
+                f"Status is one of '{config['UPDATE_ACCESS_STATUSES']}', attempting to {action} access in Arborist"
             )
 
             # assume we are always granting a user access to a resource.
             # in the future we may want to handle more use cases
             logger.debug(
                 f"username: {request.username}, policy_id: {request.policy_id}"
             )
-            success = await arborist.grant_user_access_to_policy(
-                api_request.app.arborist_client,
-                request.username,
-                request.policy_id,
-                request.resource_display_name,
-            )
+            if request.revoke:
```

Comment:
Okay, I see we are revoking here in the `/PUT` update request. Don't we (need to) perform this action in `/POST` request as well?

Reply:
there are a few things that need to be updated to allow granting access through the request creation endpoint. See PXP-8422